migrations/3_update_to_sol-0.5.js

```diff
@@ -0,0 +1,13 @@
+const IdentityFactory = artifacts.require("IdentityFactory");
+const AnchorRepository = artifacts.require("AnchorRepository");
+const PaymentObligation = artifacts.require("PaymentObligation");
+
+module.exports = function (deployer) {
```

Comment:
why not promise-chaining here?
https://javascript.info/promise-chaining

Reply:
it is promise chaining but for example this way in the last async call i can reuse the arguments from the first closure. In you example I would have to pass that around as they do not share scope